transform-plugins/src/main/java/co/cask/hydrator/plugin/Decompressor.java

```diff
@@ -104,23 +104,36 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
       pipelineConfigurer.getStageConfigurer().setOutputSchema(outputSchema);
     } catch (IOException e) {
       throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format. " +
-        e.getMessage());
+                                           e.getMessage());
+    }
+
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      for (Schema.Field field : inputSchema.getFields()) {
+        if (outSchemaMap.containsKey(field.getName()) &&
+          deCompMap.containsKey(field.getName()) && deCompMap.get(field.getName()) != DecompressorType.NONE &&
+          !Schema.Type.BYTES.equals(field.getSchema().getType())) {
+          throw new IllegalArgumentException(
+            String.format("Input field  %s should be of type BYTES to decompress. It is currently of type %s",
```

Comment:
use the same error message as above for consistency:

"Field %s must be of type bytes. It is currently of type %s"